integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/DistroXClusterFaultTolerantUpscaleTest.java

```diff
@@ -0,0 +1,221 @@
+            then = "the cluster should be AVAILABLE in a partially upscaled state")
+    public void testUpscaleWithWaitingForHostsTimingOut(MockedTestContext testContext, ITestContext testNgContext) {
+        String stack = resourcePropertyProvider().getName();
+        createDatalake(testContext);
+
+        ApiHostList apiHostList = getMockApiHostList();
+
+        DistroXTestDto currentContext = setUpContextForUpscale(testContext, stack);
+        currentContext
+                .mockCm().hosts().get().thenReturn(apiHostList, null, 200, 0, null)
+                .when(distroXClient.scale(UPSCALED_INSTANCE_GROUP_NAME, SUCCESSFUL_UPSCALE_NODE_COUNT))
+                .awaitForFlow(RunningParameter.waitForFlowFail())
+                .await(STACK_AVAILABLE, key(stack).withPollingInterval(POLLING_INTERVAL))
+                .then((tc, testDto, client) -> verifyHealthyNodeCountAfterUpscale(testDto));
+
+        currentContext.validate();
+    }
+
+    private DistroXTestDto setUpContextForUpscale(MockedTestContext testContext, String stack) {
+        return testContext
+                .given(DIX_NET_KEY, DistroXNetworkTestDto.class)
+                .given(DIX_IMG_KEY, DistroXImageTestDto.class)
+                .withImageCatalog()
+                .withImageId(IMAGE_CATALOG_ID)
+                .given(CM_FOR_DISTRO_X, DistroXClouderaManagerTestDto.class)
+                .given(CLUSTER_KEY, DistroXClusterTestDto.class)
+                .withValidateBlueprint(false)
+                .withClouderaManager(CM_FOR_DISTRO_X)
+                .given(stack, DistroXTestDto.class)
+                .withCluster(CLUSTER_KEY)
+                .withName(stack)
+                .withImageSettings(DIX_IMG_KEY)
+                .withNetwork(DIX_NET_KEY)
+                .when(distroXClient.create(), key(stack))
+                .awaitForFlow()
+                .await(STACK_AVAILABLE, key(stack).withIgnoredStatues(Set.of(Status.UNREACHABLE)));
+    }
+
+    private MinionIpAddressesResponse getMockResponseForIpAddrsSaltCall() {
+        int successfulNodeCount = SUCCESSFUL_UPSCALE_NODE_COUNT - FAILING_NODE_COUNT;
+        Map<String, JsonNode> ipAddressesForMinions = new HashMap<>();
+        try {
+            ipAddressesForMinions.put("host-192-2-0-0.ipatest.local", JsonUtil.readTree("[\"192.2.0.0\"]"));
+            for (int i = 3; i < successfulNodeCount; i++) {
+                ipAddressesForMinions.put("host-192-3-0-" + i + ".ipatest.local", JsonUtil.readTree("[\"192.3.0." + i + "\"]"));
+            }
+            for (int i = 0; i < FAILING_NODE_COUNT; i++) {
+                ipAddressesForMinions.put("host-192-3-0-" + (i + successfulNodeCount) + ".ipatest.local", JsonUtil.readTree("[\"\"]"));
+            }
+        } catch (IOException e) {
+            throw new TestFailException("The specified IP addresses could not be serialized. IOException: " + e.getMessage());
```

Comment:
Nit: `serialized` -> `deserialized` (JSON string representation to Java object).